plugins/base/frontend/build.gradle.kts

```diff
@@ -11,19 +15,34 @@ node {
     distBaseUrl.set(null as String?) // Strange cast to avoid overload ambiguity
 }
 
-val npmRunBuild = tasks.getByName("npm_run_build") {
-    inputs.dir(file("src/main"))
-    inputs.files(file("package.json"), file("webpack.config.js"))
-    outputs.dir(file("dist/"))
+val distributionDirectory = layout.projectDirectory.dir("dist")
+
+val npmRunBuild by tasks.registering(NpmTask::class) {
+    dependsOn(tasks.npmInstall)
+
+    npmCommand.set(parseSpaceSeparatedArgs("run build"))
+
+    inputs.dir("src/main")
+    inputs.files(
+        "package.json",
+        "webpack.config.js",
+    )
+
+    outputs.dir(distributionDirectory)
     outputs.cacheIf { true }
 }
```

Comment:
some minor tidying here, which hopefully makes the task more understandable.

I removed [the task-rule based creator](https://github.com/node-gradle/gradle-node-plugin/blob/642f30439f186c66e14d45bbc0c8518892e19bc5/src/main/kotlin/com/github/gradle/node/NodePlugin.kt#L67-L80) because I think it's a little less 'magic'.